notebook to demonstrate Gemma/Gemini evaluation

deep-diver opened the pull request.

deep-diver committed:
notebook to demonstrate Gemma/Gemini evaluation

sayakpaul commented on notebooks/Gemma+Gemini+Eval.ipynb:
```diff
@@ -0,0 +1,8838 @@
+{
```
<p>Can just specify <code>load_in_8bit=True</code> to <code>AutoModelForCausalLM</code> no? Or does it throw deprecation warnings?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-diver/llmops-pipeline/pull/5/discussion/#comment-1536742384'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

deep-diver replied:
It throws depreciation warning and it recommended to use quantization_config in the future instead

sayakpaul commented on notebooks/Gemma+Gemini+Eval.ipynb:
```diff
@@ -0,0 +1,8838 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #14.        gen_input = tokenizer.apply_chat_template(messages, return_tensors="pt", add_generation_prompt=True).to(model.device)
</pre><p> </p><p>Since <code>device_map="auto"</code> we don't need to do the device placement manually :)  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-diver/llmops-pipeline/pull/5/discussion/#comment-1536742385'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

deep-diver replied:
Didn't know this is applicable to the tokenizer!! Thanks for the comment :)

sayakpaul commented on notebooks/Gemma+Gemini+Eval.ipynb:
```diff
@@ -0,0 +1,8838 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #4.    def find_json_snippet(raw_snippet):
</pre><p>What's the purpose of this thing? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-diver/llmops-pipeline/pull/5/discussion/#comment-1536742386'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

deep-diver replied:
Parsing JSON bits from the output. Sometimes, it emits with and without ``` or ```json. To make sure if there is passable JSON, I am manually finding that out with this simple method. Have been using it for a while, and it worked on all cases well

sayakpaul commented on notebooks/Gemma+Gemini+Eval.ipynb:
```diff
@@ -0,0 +1,8838 @@
+{
```
<p>If you have received Gemini 1.5 API access, let's include that too? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-diver/llmops-pipeline/pull/5/discussion/#comment-1536742387'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

deep-diver replied:
Not yet! maybe yes, but Gemini 1.5 Pro takes a long time for a single inference (at least 30 seconds I believe)

sayakpaul replied:
Oh wow!

sayakpaul commented:
Thanks for the answers. 

In `tokenizer.decode()` we should add `tokenizer.decode(skip_special_tokens=True)` as well.

deep-diver committed:
skip_special_tokens=True

deep-diver commented:
@sayakpaul 

Addressed your comments. Specifically:
- added `skip_special_tokens=True` in `decode()`, then split by the delimiter which is `assistant\n` in this case.
- removed `.to(model.device)` after the `apply_chat_template()` as you suggested!

sayakpaul commented:
Thanks! The first resolution should help us get rid of the manual eos splitting we were doing. 

Feel free to merge this!
